### PR TITLE
update kustomize roadmap 23-24

### PR DESCRIPTION
**Pull request by natasha41575**
Defining the kustomize roadmap for the next 1 year

/cc @annasong20 @koba1t @varshaprasad96

### CODE REVIEW

**Review by varshaprasad96 — COMMENTED** [review]
Going through the doc, the goals seem well classified based on the cohort training, contributing well to the overall health of the project. Looks good to me!

**Comment by annasong20** [comment]
Very clear and well-explained!!
/lgtm

**Comment by koba1t** [comment]
I think this is a good plan!
/lgtm

**Review by seans3 — APPROVED** [review]
I think this looks good. Thanks updating this roadmap to the current/next year. I'll give some time for Maciej to have a look.

**Review by soltysh — APPROVED** [review]
A few minor wording nits and link suggestions. Overall lgtm - thank you!

File: ROADMAP.md [hunk 1]
```diff
@@ -1,112 +1,173 @@
...
-Presented at the [January 26, 2022, SIG-CLI meeting](https://youtu.be/l2plzJ9MRlk?t=1321)
+This document describes the items that we hope to make progress on over the next
+1 year (H2 2023 and H1 2024). Take this roadmap as more of what we hope to achieve
+rather than what we promise to achieve, as some items in this doc are highly dependent
+on the success that we have on-ramping new contributors to the project, and other
+items will depend on external contributions, which can vary.
 
-kustomize maintainers: @knverey, @natasha41575
+If you are interested in contributing to a particular area, you can look through
+the project board for that area and assign yourself to one of the issues. It is
```
[the remark this answers is not in the document]
Comment (natasha41575) [reply ROADMAP.md:10]: Thanks! I like the idea of linking all the project boards. 

I think https://cli.triage.k8s.io/s/kustomize might be a little confusing to a new contributor reading the roadmap, but I added it as a link to the triage guide instead.

File: ROADMAP.md [hunk 2]
```diff
@@ -1,112 +1,173 @@
...
-Also very valuable to the project:
+Effort: Medium
 
-- [Improve the release process](https://github.com/kubernetes-sigs/kustomize/issues/3952) to support regular biweekly releases [PROJECT](https://github.com/kubernetes-sigs/kustomize/projects/7)
-- Release sigs.k8s.io/kustomize/api v1.0.0 [PROJECT](https://github.com/kubernetes-sigs/kustomize/projects/5)
-  - [Reduce the public surface of the API module](https://github.com/kubernetes-sigs/kustomize/issues/3942)
-  - [Vendor all transitive deps](https://github.com/kubernetes-sigs/kustomize/issues/3706). Since kustomize is in kubectl, we must do as kubectl does to manage deps, exposing new transitive deps in code review.
-- Project administration
-  - [Rename master branch to main](https://github.com/kubernetes-sigs/kustomize/issues/3954)
+In the past, when the leads have left in various kubernetes projects, it
```
[the remark this answers is not in the document]
Comment (natasha41575) [reply ROADMAP.md:33]: Done.

File: ROADMAP.md [hunk 3]
```diff
@@ -1,112 +1,173 @@
...
-- Kustomize performance investigations/improvements [PROJECT](https://github.com/kubernetes-sigs/kustomize/projects/13)
-- [Support generic resource references in name reference tracking](https://github.com/kubernetes-sigs/kustomize/issues/3418)
-- [KEP 4267: retain the resource origin and transformer data in annotations](https://github.com/kubernetes-sigs/kustomize/pull/4267)
+In order to make progress on kustomize goals in the future, we need to increase the
+level of staffing on kustomize. We should leverage community contributions to keep kustomize
+healthy and making progress.
 
-Secondary priorities:
+The primary means in which we will try to find new kustomize contributors is through the new kustomize
+maintainer training cohort. We will lead a group of ~20 kubernetes community members through a 3-6 month
```
[the remark this answers is not in the document]
Comment (natasha41575) [reply ROADMAP.md:59]: Great idea, done.

**Review by soltysh — APPROVED** [review]
/lgtm

**Pull request merged**